internal/security/config/command/proxy/tls/command.go

```diff
@@ -61,9 +78,208 @@ func NewCommand(
+		"certificates", certId}, "/")
+	c.loggingClient.Info(fmt.Sprintf("deleting tls certificate on the endpoint of %s", delCertKongURL))
+	req, err := http.NewRequest(http.MethodDelete, delCertKongURL, http.NoBody)
+	if err != nil {
+		return fmt.Errorf("Failed to prepare request to delete Kong tls cert: %w", err)
+	}
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("Failed to send request to delete Kong tls cert: %w", err)
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusNoContent:
+		c.loggingClient.Info("Successfully deleted Kong tls cert")
+	case http.StatusNotFound:
+		// not able to find this certId but should be ok to proceed and post a new certificate
+		c.loggingClient.Warn(fmt.Sprintf("Unable to delete Kong tls cert because the certificate Id %s not found", certId))
+	default:
+		return fmt.Errorf("Delete Kong tls certificate request failed with code: %d", resp.StatusCode)
+	}
+	return nil
+}
+
+func (c *cmd) postKongTLSCertificate(certKeyPair *bootstrapConfig.CertKeyPair) error {
+	postCertKongURL := strings.Join([]string{c.configuration.KongURL.GetProxyBaseURL(),
+		"certificates"}, "/")
+	c.loggingClient.Info(fmt.Sprintf("posting tls certificate on the endpoint of %s", postCertKongURL))
+
+	form := url.Values{
+		"cert": []string{certKeyPair.Cert},
+		"key":  []string{certKeyPair.Key},
+		"snis": getServerNameIndications(c.serveNameIndictionsList),
+	}
+
+	formVal := form.Encode()
+	req, err := http.NewRequest(http.MethodPost, postCertKongURL, strings.NewReader(formVal))
+	if err != nil {
+		return fmt.Errorf("Failed to prepare request to post Kong tls cert: %w", err)
+	}
+
+	req.Header.Add(clients.ContentType, common.UrlEncodedForm)
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("Failed to send request to post Kong tls cert: %w", err)
+	}
+	defer resp.Body.Close()
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("Failed to read response body: %v", err)
+	}
+
+	switch resp.StatusCode {
+	case http.StatusCreated:
+		c.loggingClient.Info("Successfully posted Kong tls cert")
+	case http.StatusBadRequest:
+		return fmt.Errorf("BadRequest as unable to post Kong tls cert due to error: %s", string(responseBody))
+	default:
+		return fmt.Errorf("Post Kong tls certificate request failed with code: %d", resp.StatusCode)
+	}
+	return nil
+}
+
+func getServerNameIndications(snisList string) []string {
+	// this will parse out the internal default server name indications and extra ones from input if given
+	var snis []string
+	snis = append(snis, parseAndTrimSpaces(builtinSNISList)...)
+
+	uniqueSnisMap := make(map[string]bool)
+	for _, s := range snis {
+		uniqueSnisMap[s] = true
+	}
+
+	if len(snisList) > 0 {
+		splitSnis := parseAndTrimSpaces(snisList)
+		for _, sni := range splitSnis {
+			if _, exists := uniqueSnisMap[sni]; !exists {
+				uniqueSnisMap[sni] = true
+				snis = append(snis, sni)
+			}
+		}
+	}
```

Comment:
This algorithm is confusing.  There is no need for the existence test.

Suggest:
create the map
iterate builtin sni list and set key of map
iterate user-supplied sni list and set key of map (potential overwrite, which is harmless)
iterate keys of resulting map and build the array

Reply:
Also if the builtin sni list was an array to start with, it could also save some extra code.

Reply:
Nevermind the above.  Arrays aren't compatible with the configuration provider.  Perhaps default SNIs should come from configuration.toml as a string vs an array.

Reply:
currently there is no toml file for this utility. so you suggest creating one? it sounds to me it's kind of overdone just for that single entry

Reply:
Just thinking out loud.  You are probably right.